plugins/SolidView/SolidView.py

```diff
@@ -98,6 +139,50 @@ def beginRendering(self):
 
         self._checkSetup()
 
+        if not CuraApplication.getInstance().getPreferences().getValue(self._show_xray_warning_preference):
+            self._xray_error_image = None
+            self._xray_shader = None
+            self._xray_composite_shader = None
+            if self._composite_pass and 'xray' in self._composite_pass.getLayerBindings():
+                self._composite_pass.setLayerBindings(self._old_layer_bindings)
+                self._composite_pass.setCompositeShader(self._old_composite_shader)
+                self._old_layer_bindings = None
+                self._old_composite_shader = None
+                self._xray_warning_message.hide()
+        else:
+            if not self._xray_error_image:
+                self._xray_error_image = OpenGL.getInstance().createTexture()
+                texture_file = "xray_error.png"
+                try:
+                    self._xray_error_image.load(Resources.getPath(Resources.Images, texture_file))
+                except FileNotFoundError:
+                    Logger.log("w", "Unable to find xray error texture image [%s]", texture_file)
+
+            if not self._xray_shader:
+                self._xray_shader = OpenGL.getInstance().createShaderProgram(Resources.getPath(Resources.Shaders, "xray.shader"))
+
+            if not self._xray_composite_shader:
+                self._xray_composite_shader = OpenGL.getInstance().createShaderProgram(Resources.getPath(Resources.Shaders, "xray_composite.shader"))
+                theme = Application.getInstance().getTheme()
+                self._xray_composite_shader.setUniformValue("u_background_color", Color(*theme.getColor("viewport_background").getRgb()))
+                self._xray_composite_shader.setUniformValue("u_outline_color", Color(*theme.getColor("model_selection_outline").getRgb()))
+                self._xray_composite_shader.setTexture(3, self._xray_error_image)
+
+            if not self._composite_pass or not 'xray' in self._composite_pass.getLayerBindings():
+                # Currently the RenderPass constructor requires a size > 0
+                # This should be fixed in RenderPass's constructor.
+                self._xray_pass = XRayPass.XRayPass(1, 1)
+
+                self.getRenderer().addRenderPass(self._xray_pass)
+
+                if not self._composite_pass:
+                    self._composite_pass = self.getRenderer().getRenderPass("composite")
+
+                self._old_layer_bindings = self._composite_pass.getLayerBindings()
+                self._composite_pass.setLayerBindings(["default", "selection", "xray"])
+                self._old_composite_shader = self._composite_pass.getCompositeShader()
+                self._composite_pass.setCompositeShader(self._xray_composite_shader)
```

Comment:
I think it's better to do all this in `self._checkSetup()`, right?

Reply:
You might be right there. I don't know about that function. I haven't considered it while writing the code. It didn't figure in the code for the xray view, so I haven;t looked at it.

Reply:
I've moved it into that function: a4fbf3c613bfe8f91b1bb63b472e2f9ff22b69bb